✨ feat: SJRA-274 Integrate clinical data model

kritchie opened the pull request:
https://d3b.atlassian.net/browse/SJRA-274

## Context

With https://d3b.atlassian.net/browse/SJRA-265  completed, the clinical data model was made available locally (with synthetic data) for development. 

This data model defines concepts such as cases , sequencing_experiment , task , etc… 

We need to integrate the existing model in StarRocks with this new model in order to orchestrate the ETL dags. 

## Contribution

The main contributions of this PR are:

- Adds a new dag `import_radiant.py` that can be scheduled periodically to import partitions. 
- Renames `sequencing_experiment` into `staging_sequencing_experiment` 
- Adds the `staging_external_sequencing_experiment` view to expose clinical sequencing experiments in StarRocks.
- Adds the `staging_sequencing_experiment_delta` view that computes which new sequencing experiments need to be processed.
- Adds partition and priority assigner classes to handle the partitioning logic. 

### How do we compute partition keys ? 

![img_1](https://github.com/user-attachments/assets/16d61730-0f8b-4fd5-bbe2-1699dcd293ef)

### How do we determine what will be processed in a DAG run ? 

![img_2](https://github.com/user-attachments/assets/3ae46ea2-0a1e-452f-ad17-0d08cca3575e)

jecos commented on tests/integration/dags/sql/test_sequencing_experiment_select_delta.py:
Can we have a test for same task imported twice but with different update date? Ex. : a task has been updated in sequencing experiment postgres table.

kritchie commented:
@jecos @celinepelletier 

Added in the last commit: 

- Renamed `sequencing_experiment` -> `staging_sequencing_experiment` (Table)
- Renamed `staging_sequencing_experiment` -> `staging_external_sequencing_experiment` (View)
- Added test case to validate updated sequencing experiment is taking re-imported
